controllers/etcdcluster_controller.go

```diff
@@ -62,22 +69,44 @@ func headlessServiceForCluster(cluster *etcdv1alpha1.EtcdCluster) *v1.Service {
 	}
 }
 
-// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters,verbs=get;list;watch
-// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdclusters/status,verbs=get;update;patch
-// +kubebuilder:rbac:groups=core,resources=services,verbs=get;list;watch;create
-// +kubebuilder:rbac:groups=etcd.improbable.io,resources=etcdpeers,verbs=get;list;watch;create
+// updateStatus updates the EtcdCluster resource's status to be the current value of the cluster.
+func (r *EtcdClusterReconciler) updateStatus(
+	ctx context.Context,
+	name types.NamespacedName,
+	cluster *etcdv1alpha1.EtcdCluster,
+	members []etcdclient.Member,
+	reconcilerEvent event.EtcdClusterReconcilerEvent) error {
 
-func (r *EtcdClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
+	log := r.Log.WithValues("cluster", name)
 
-	log := r.Log.WithValues("cluster", req.NamespacedName)
+	if members != nil {
+		cluster.Status.Members = make([]etcdv1alpha1.EtcdMember, len(members))
+		for i, member := range members {
+			cluster.Status.Members[i] = etcdv1alpha1.EtcdMember{
+				Name: member.Name,
+				ID:   member.ID,
+			}
+		}
 
-	cluster := &etcdv1alpha1.EtcdCluster{}
-	if err := r.Get(ctx, req.NamespacedName, cluster); err != nil {
-		log.Error(err, "unable to fetch EtcdCluster")
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		if err := r.Client.Status().Update(ctx, cluster); err != nil {
+			log.Error(err, "Unable to update status field on EtcdCluster")
+			return err
+		}
 	}
+	return nil
+}
+
+func (r *EtcdClusterReconciler) reconcile(
+	ctx context.Context,
+	name types.NamespacedName,
+	members []etcdclient.Member,
+	cluster *etcdv1alpha1.EtcdCluster,
```

Comment:
Do we need to pass the name *and* the cluster? Can't we use `Cluster.Spec.Name` ?